### PR TITLE
Store: Replace thumb placeholder with WC generic placeholder image

### LINKED ISSUE #21268: Store: Replace thumb placeholder with WC generic placeholder
Right now in various parts of the UI like products, variations, and product categories, we use a gray box as a placeholder for when we don't have a thumbnail. This can be confusing or look like things are still loading in.

Per https://github.com/Automattic/wp-calypso/pull/21227#issuecomment-355104365

> The image that actually gets added to the front end is the WC generic placeholder image, which is about to updated in the next version of WC. I'm thinking we should just use that?

### PR DESCRIPTION
**Pull request by justinshreve**
Fixes #21268.

This updates the little gray thumbnail placeholder (when a product or category doesn't have an image) with the generic placeholder image that is being used in newest version of WooCommerce.

To Test, verify that the image placeholder is used on the following screens:
* `http://calypso.localhost:3000/store/order/:site` - The "add products" dialog
* `http://calypso.localhost:3000/store/promotion/:site` - The "specific product" card
* `http://calypso.localhost:3000/store/products/:site` - The product listing page
* `http://calypso.localhost:3000/store/products/categories/:site` - The product category listing page

screen shot 2018-01-25 at 11 17 53 am

screen shot 2018-01-25 at 11 18 05 am

### CODE REVIEW

**Comment by allendav** [comment]
Would you consider having a single component for images that accepts props like className and src, and lacking a src, renders the placeholder?  Might be DRYer that way

**Comment by justinshreve** [comment]
@allendav Good call! That cleaned things up a bit, especially the CSS. https://github.com/Automattic/wp-calypso/pull/21841/commits/7bfa8b03d4afdbda61d062c604ab95679ed11cb6

**Comment by jameskoster** [comment]
I love that image but I wonder if we shouldn't use something from the new wp.com brand illustrations for this? What do you think @kellychoffman ?

File: client/extensions/woocommerce/components/image-thumb/index.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+	const imageClasses = classNames( 'image-thumb', {
+		hasImage: src,
+	} );
+	const placeholderClasses = classNames( 'image-thumb__placeholder' );
+
+	if ( ! src ) {
+		return (
+			<div className={ imageClasses }>
+				{' '}
```
Comment (ryelle) [inline client/extensions/woocommerce/components/image-thumb/index.js]: Why do we need the spaces here?

**Comment by kellychoffman** [comment]
I was thinking that it would match what was shown on the front end.

File: client/extensions/woocommerce/components/image-thumb/index.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	const style = {
+		width,
+		height,
+		minHeight: height,
+	};
+
+	const imageClasses = classNames( 'image-thumb', {
+		hasImage: src,
+	} );
+	const placeholderClasses = classNames( 'image-thumb__placeholder' );
```
Comment (ryelle) [inline client/extensions/woocommerce/components/image-thumb/index.js]: This doesn't need the `classNames` call - could you just put this directly on the div?

File: client/extensions/woocommerce/components/image-thumb/index.js (hunk 1 above)
Comment (justinshreve) [reply client/extensions/woocommerce/components/image-thumb/index.js]: Uhh, I'm actually not sure how those ended up there but they are not needed. Thanks for catching.

**Comment by justinshreve** [comment]
Thanks for the review @ryelle. I added an optional placeholder attribute and addressed your other notes.

File: client/extensions/woocommerce/components/image-thumb/index.js (hunk 1 above)
Comment (ryelle) [reply client/extensions/woocommerce/components/image-thumb/index.js]: 👍 i think the prettier formatting adds them when it wraps lines of text, maybe that's what happened here.

**Pull request merged**